components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/DefaultStepBasedSequenceHandler.java

```diff
@@ -773,4 +775,33 @@ protected void resetAuthenticationContext(AuthenticationContext context) throws
         context.setCurrentAuthenticator(null);
     }
 
+    /**
+     * Returns the local claim uri that is mapped for the IdP role claim uri configured.
+     * If no role claim uri is configured for the IdP returns the local role claim 'http://wso2.org/claims/role'.
+     *
+     * @param externalIdPConfig IdP configurations
+     * @return local claim uri mapped for the IdP role claim uri.
+     * @throws FrameworkException
+     */
+    protected String getLocalClaimUriMappedForIdPRoleClaim(ExternalIdPConfig externalIdPConfig) throws
+            FrameworkException {
+        // get external identity provider role claim uri.
+        String idpRoleClaimUri = externalIdPConfig.getRoleClaimUri();
+        if (StringUtils.isNotBlank(idpRoleClaimUri)) {
+            // Iterate over IdP claim mappings and check for the local claim that is mapped for the remote IdP role
+            // claim uri configured.
+            ClaimMapping[] idpToLocalClaimMapping = externalIdPConfig.getClaimMappings();
+            if (idpToLocalClaimMapping != null && idpToLocalClaimMapping.length > 0) {
+                for (ClaimMapping mapping : idpToLocalClaimMapping) {
+                    if (mapping.getRemoteClaim() != null && idpRoleClaimUri.equals(mapping.getRemoteClaim()
+                            .getClaimUri())) {
+                        return mapping.getLocalClaim().getClaimUri();
```

Comment:
Please check and verify whether mapping.getLocalClaim() can be null

Reply:
If IdP claim mappings are populated it won't be null. Always local claims are available but the possibility is remote claims can be null.
Moreover, if local claims are null in the IdP claim mapping it won't come up to this point.